ban/http/resources.py

```diff
@@ -16,38 +16,116 @@ class BaseCollection:
     DEFAULT_LIMIT = 20
     MAX_LIMIT = 100
 
-    def get_limit(self, req):
-        return min(int(req.params.get('limit', self.DEFAULT_LIMIT)),
-                   self.MAX_LIMIT)
+    def get_start(self, req, count):
+        """Get start value from request.
+
+        Keyword arguments:
+            req -- Request.
+
+        Business rules:
+            if start == not defined then start = 0.
+            if start < 0            then start = 0.
+            if start > count        then start = 0.
+            else start = start.
+        """
+        try:
+            start = int(req.params.get('start', 0))
+        except (ValueError, TypeError):
+            return 0
+        else:
+            if (start <= 0):
+                start = 0
+            if (start > count):
+                start = 0
+            else:
+                start = start
+
+            return start
+
+    def get_limit(self, req, count):
+        """Get limit value from request.
+
+        Keyword arguments:
+            req -- Request.
+
+        Business rules:
+            if limit == not defined then limit = count.
+            if limit < 0            then limit = count.
+            if limit > count        then limit = count.
+            else limit = limit.
+
+        """
+        try:
+            limit = int(req.params.get('limit', count))
+        except (ValueError, TypeError):
+            return 0
+        else:
+            if (limit <= 0):
+                limit = count
+            if (limit > count):
```

Comment:
We are changing the default limit from `20` to `0` basically. I'm not sure we want that.